#234207 - ProcessManagement step prioritization change

VishnuReghuMDSL opened the pull request:
- Fixed the issue where processing the 1st step of the 1st pipeline - the 2nd step of the 1st pipeline goes to the back of the queue (after the 1st step of the 500th pipeline).
- Added a while loop to go through each step in the pipeline before moving to next pipeline, so all steps in that pipeline will be executed first
- If there was an error while executing the step, the while loop was continously looking for the failed step hence all other workflows were not starting as it was waiting for this never ending while loop
- Hence checking for StartTime is null when taking list of second step


**Describe the change**
A clear and concise description of what the change is. Any PR submitted without a description of the change will not be reviewed.

**Describe your implementation or design**
How did you go about implementing the change?

**Tests**
Did you cover your changes with tests?

**Breaking change**
Do you changes break compatibility with previous versions?

**Additional context**
Any additional information you'd like to provide?

Commit:
#234207 - ProcessManagement step prioritization change
- Fixed the issue where processing the 1st step of the 1st pipeline - the 2nd step of the 1st pipeline goes to the back of the queue (after the 1st step of the 500th pipeline).
- Added a while loop to go through each step in the pipeline before moving to next pipeline, so all steps in that pipeline will be executed first
- If there was an error while executing the step, the while loop was continously looking for the failed step hence all other workflows were not starting as it was waiting for this never ending while loop
- Hence checking for StartTime is null when taking list of second step

DaveFuller-Calero commented on src/WorkflowCore/WorkflowCore.csproj:
```diff
@@ -6,15 +6,22 @@
     <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
     <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
     <Description>Workflow Core is a light weight workflow engine targeting .NET Standard.</Description>
+	<PublishNupkgLibraryCommand>nuget.exe push -Source https://intelligentcomms.pkgs.visualstudio.com/_packaging/calero-mdsl/nuget/v3/index.json -ApiKey az</PublishNupkgLibraryCommand>
```
Do we actually need the changes that are in this file?
Our github is currently open, so anyone could view these api keys.
If we need this then perhaps we need to make our github private.

VishnuReghuMDSL replied:
Removed unnecessary tag. We can always add in local .csproj before pushing the nuget package.

DaveFuller-Calero commented on src/WorkflowCore/MDSL.GitInfo.props:
```diff
@@ -0,0 +1,30 @@
+﻿<!-- 
```
Do we actually need this file?

VishnuReghuMDSL replied:
We will keep this file for versioning as Hydra.STL.NugetPackaging uses this file to version the nuget package.

DaveFuller-Calero reviewed (commented):
The main WorkflowExecutor.cs change is what we need, and looks fine (it looks like a lot changed, but it's mostly whitespace changes).
I'm not sure if we need the changes from the other 2 files. I'd like to keep changes to a minimum if these aren't needed, to help avoid conflicts - but if they're required for us I think they're fine.

Commit:
#234207 - ProcessManagement step prioritization change
- Changed Author name
- Added additional properties
- Removed unnecessary properties
